src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -197,16 +208,31 @@ def build_graph_from_python_source(self, python_code: str) -> list[DependencyPro
                 problems.append(problem)
         return problems
 
+    @classmethod
+    def convert_magic_commands(cls, python_code: str):
+        lines = python_code.split("\n")
+        for i, line in enumerate(lines):
+            if not line.startswith("%"):
+                continue
+            content = base64.b64encode(line.encode())  # use base64 to simplify escaping
+            lines[i] = f"magic_command('{content.decode('ascii')}')"
```

Comment:
yep, this is roughly how jupyter implements it. `%pip x` is translated into `get_ipython().run_cell_magic('pip', 'x')`

look at `IPython.core.tests.test_inputsplitter.CellMagicsCommon` and surrounding classes

Reply:
noted